File: src/main/ATOMRuntime.java
package main;

import java.io.File;
import java.util.ArrayList;
import java.util.HashMap;

public class ATOMRuntime {

    public static HashMap<String, ATOMValue> globalVars = new HashMap<>();

    public static void reset() {
        globalVars = new HashMap<>();
    }

    public static ATOMValue processInput(String input) {
        return ATOMElement.fromTemplate(new ATOMTemplate(input)).eval();
    }

    // BUNDLER: BEGIN IGNORE
    public static ATOMValue processFile(File f) throws Exception {
        ATOMTemplate atom = new ATOMTemplate();
        atom.importFile(f);
        ATOMScope program = (ATOMScope) ATOMElement.fromTemplate(atom);
        return program.eval();
    }
    // BUNDLER: END IGNORE

    public static ATOMValue getGlobalVar(String name) {
        if (globalVars.containsKey(name)) {
            return globalVars.get(name);
        }
        return ATOMValue.NULL();
    }

    public static void injectVariable(String name, ATOMValue value) {
        globalVars.put(name, value);
    }

}

File: web_interpreter/atom/ATOMRuntime.js
import ATOMElement from './ATOMElement.js';
import ATOMTemplate from './ATOMTemplate.js';
import ATOMValue from './ATOMValue.js';

export default class ATOMRuntime {

  static globalVars = {};
  static stack = [];

  static reset() {
    ATOMRuntime.globalVars = {};
  }

  static getIndexedVar(index) {
    return ATOMRuntime.stack[ATOMRuntime.stack.length-(index+1)];
  }

  static pushIndexedVar(val) {
    ATOMRuntime.stack.push(val);
  }

  static popIndexedVar() {
    let val = ATOMRuntime.stack.splice(ATOMRuntime.stack.length-1, 1);
    return val[0];
  }

  static processInput(input) {
    return ATOMElement.fromTemplate(new ATOMTemplate(input)).eval();
  }

  static getGlobalVar(name) {
    if (ATOMRuntime.globalVars[name]!=null) {
      return ATOMRuntime.globalVars[name];
    }
    return ATOMValue.NULL();
  }

  static injectVariable(name, value) {
    ATOMRuntime.globalVars[name] = value;
  }

}